pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -247,6 +259,41 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *
 	return nil
 }
 
+func finalizeIfNeeded(ctx context.Context, stream *imagev1.ImageStream, srcClusterName string, clients map[string]ctrlruntimeclient.Client) error {
+	finalizerSet := sets.NewString(stream.Finalizers...)
+	if finalizerSet.Has(ControllerName) {
+		for clusterName, client := range clients {
+			if clusterName == srcClusterName {
+				continue
+			}
+			// populate deleting to all clusters except srcClusterName
+			if err := client.Delete(ctx, &imagev1.ImageStream{
```

Comment:
This will need to delete the finalizer if present which can happen if we do some back and forth syncing

Reply:
Updated.
I am curious of what would happen otherwise.

`finalize` on A -> `delete` B -> `finalize` on B -> `delete` A again (if A is still on the system)?